independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/clientproxy/bridgemethod/SameDescriptorDifferentReturnTypeMethodTest.java

```diff
@@ -15,20 +15,23 @@
 public class SameDescriptorDifferentReturnTypeMethodTest {
 
     @RegisterExtension
-    public ArcTestContainer container = new ArcTestContainer(LoopProducer.class, Loop.class, SuperLoop.class);
+    public ArcTestContainer container = new ArcTestContainer(LoopProducer.class, Loop.class, SuperLoop.class,
+            EvenMoreSuperLoop.class, ComplexLoop.class, Complex.class);
 
     @Test
     public void testProxy() throws IOException {
         Serializable ret = Arc.container().instance(SuperLoop.class).get().next();
         assertEquals(9, ret);
+        Object complexRet = Arc.container().instance(EvenMoreSuperLoop.class, new Complex.Literal()).get().next();
```

Comment:
Are you sure this is testing the bridge method?

This is testing the `Object()` method, pretty sure. Not the `Integer()` override. Now, granted, I can't remember which of the two is the bridge method at this instant :)
Or indeed, which is the one that is causing issues.

But one thing for sure, this is not testing both methods, right?
Not sure it has to, either.

Reply:
In fact, I don't think we can test the anonymous class here because the bean types of the producer only include `ComplexLoop`,  `SuperLoop` and `EvenMoreSuperLoop`.

Reply:
You could do that if you turned it into a non-anonymous class with a subinterface. But not sure it's important. I'll let you judge.